x/stakeibc/keeper/icacallbacks_delegate.go

```diff
@@ -92,9 +92,9 @@ func DelegateCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, ack
 	// Update delegations on the host zone
 	for _, splitDelegation := range delegateCallback.SplitDelegations {
 		hostZone.TotalDelegations = hostZone.TotalDelegations.Add(splitDelegation.Amount)
-		success := k.AddDelegationToValidator(ctx, hostZone, splitDelegation.Validator, splitDelegation.Amount, ICACallbackID_Delegate)
-		if !success {
-			return errorsmod.Wrapf(types.ErrValidatorDelegationChg, "Failed to add delegation to validator")
+		err := k.AddDelegationToValidator(ctx, hostZone, splitDelegation.Validator, splitDelegation.Amount, ICACallbackID_Delegate)
```

Comment:
Shouldn't we increment TotalDelegations here? If so, seems like a good reason to nest that update inside `AddDelegationToValidator`

Reply:
If so, we should add a check on TotalDelegations to the `TestAddDelegationToValidator` unit test too

Reply:
yeah see my other message, it seems a bit trickier, but I agree it would be safer to update both, so I'll give it another look

Reply:
replied in the other comment